cli/azd/pkg/spin/run.go

```diff
@@ -56,10 +76,31 @@ func RunWithUpdater(prefix string, runFn RunWithUpdateFunc, finalFuncs ...func(*
 		}
 	}()
 
-	result = runFn(func(newPrefix string) {
-		if newPrefix != "" {
-			spin.Prefix(newPrefix)
-		}
+	logMutex := sync.Mutex{}
+
+	result = runFn(&SpinnerUpdater{
+		UpdatePrefix: func(newPrefix string) {
+			if newPrefix != "" {
+				spin.Prefix(newPrefix)
+			}
+		},
+		LogMessage: func(message string) {
+			if message != "" {
+				defer logMutex.Unlock()
+
+				logMutex.Lock()
+				// Ignore error returned (which can only happen if spinner is not running)
+				// We control the spinner's state, so the spinner is guaranteed to be running
+				// nolint:errcheck
+				spin.Stop()
```

Comment:
The go way to ignore error is by assigning to `_`
```suggestion
				_ = spin.Stop()
```